src/mixer/basetrackplayer.cpp

```diff
@@ -335,3 +345,30 @@ void BaseTrackPlayerImpl::setupEqControls() {
     m_pSpeed = new ControlObjectSlave(group, "rate", this);
     m_pPitchAdjust = new ControlObjectSlave(group, "pitch_adjust", this);
 }
+
+void BaseTrackPlayerImpl::slotPassthroughEnabled(double v) {
+    bool configured = m_pInputEnabled->toBool();
+    bool passthrough = v > 0.0;
+
+    // Warn the user if they try to enable passthrough on a player with no
+    // configured input.
+    if (!configured && passthrough) {
+        m_pPassthroughEnabled->set(0.0);
```

Comment:
Maybe this is a hint that some code is missplaced.
maybe slotPassthroughEnabled can emmit only the signal 

and void EngineDeck::slotPassingToggle(double v) can be turned into a 
connectValueChangeRequest and do the checking.